Strimzi Kafka Cluster deployment liveness test

mtarng opened the pull request:
Basic liveness test for a Strimzi Kafka Cluster. Creates a new topic; creates a producer & consumer using kafka client and verifies messages. Currently ignoring the CRD to create topics as editing the topic name isn't very flexible.

Probably need to expand edge cases, error catching, and documentation. This is the first pass until we have a better idea of what's needed for end to end testing.

We can also consider merging this into the `/test` directory and creating a joint README for all tests.

Addresses https://github.com/microsoft/bedrock/issues/520

mtarng committed:
Strimzi Kafka Cluster deployment liveness test

andrebriggs commented on livetest/livetest.sh:
```diff
@@ -0,0 +1,95 @@
+sleep 5s
+
+# Kafka client SSL Components
+setup_kafka_client_ssl () {
```
@mtarng we don't seem to be using this. Especially this this test is using `kubectl exec` We can probably get rid of this method and reference until we have a test that using ingress

andrebriggs reviewed (changes requested):
This is a good start. The main question is are are overcomplicating this first PR by including  `setup_ssl.sh`?

sayar commented on livetest/livetest.sh:
```diff
@@ -0,0 +1,95 @@
+cat $MESSAGE_INPUT_FILE
+
+# Create messages via console producer
+kubectl exec -n kafka -i kafkaclient-0 -- bin/kafka-console-producer.sh --broker-list kcluster-kafka-brokers:9092 --topic $TESTING_TOPIC < $MESSAGE_INPUT_FILE
```
Does this work with strimzi? If it does, somehow our network policies are too permissive because it shouldn't connect without ssl.

sayar reviewed (changes requested):
There is a lot of duplicate code as with the test folder. We should merge the two together. Also, I need to investigate how you can connect on the 9092 port from the kafkaclient. It shouldn't have access.

mtarng committed:
DRYing livetest

mtarng commented:
Moved to `/test` and utilizing shared components with other tests. Also removed unused SSL component for now, can add back if/when we need to support it.

andrebriggs commented on test/livetest.sh:
```diff
@@ -4,23 +4,11 @@
 
 # Deploy kafka client to use as producer and consumer
 # TODO: figure out if we can deploy client without topic and users first?
+kubectl create secret generic -n kafka mirror-maker-cluster-ca-cert --from-literal=ca.crt=sample-faked-cert
```
@mtarng is this necessary for the test in it's current state?

mtarng replied:
Yeah. The kafka client configuration expects a secret for mirror maker, and since I'm piggy-backing off the same client, we need a dummy secret.

andrebriggs replied:
@mtarng Can we make t more client in the comment that we only add this because the test expects it? Also that is it a dummy cert? Otherwise this PR is fine

mtarng committed:
Clarifying comments in test

mtarng commented:
@sayar any other comments?
